config/config.go

```diff
@@ -584,18 +598,57 @@ type MarathonSDConfig struct {
 	XXX map[string]interface{} `yaml:",inline"`
 }
 
+// KubernetesSDConfig is the configuration for Kubernetes service discovery.
+type KubernetesSDConfig struct {
+	Server           string   `yaml:"server"`
+	NodeMetricsPort  int      `yaml:"node_metrics_port,omitempty"`
+	InCluster        bool     `yaml:"in_cluster,omitempty"`
+	BearerTokenFile  string   `yaml:"bearer_token_file,omitempty"`
+	Username         string   `yaml:"username,omitempty"`
+	Password         string   `yaml:"password,omitempty"`
+	Insecure         bool     `yaml:"insecure,omitempty"`
+	CertFile         string   `yaml:"cert_file,omitempty"`
+	KeyFile          string   `yaml:"key_file,omitempty"`
+	CAFile           string   `yaml:"ca_file,omitempty"`
+	RetryInterval    Duration `yaml:"retry_interval,omitempty"`
+	RequestTimeout   Duration `yaml:"request_timeout,omitempty"`
+	DiscoverServices bool     `yaml:"discover_services,omitempty"`
+	ScrapeKey        string   `yaml:"scrape_key,omitempty"`
```

Comment:
These key options all seem to be doing a more primitive version of relabelling, and you're hardcoding that the user has a particular deployment where particular labels must be set. Anything we provide with Prometheus needs to be generic enough to work for all reasonable deployment models.

It's not clear to me that this supports a "scrape everything" model, and scheme/port/path should be done with relabelling.